### DISCUSSION TITLE
Topology Optimization with DASimpleTFoam - Material interpolation

### DISCUSSION THREAD

**Discussion by schewef** [mdolab/dafoam#469] [General]
Dear @friedenhe,

I am using the "alphaPorosity" term implemented in DASimpleTFoam for Topology Optimization of a Conjugate Heat Transfer problem.
Currently the alphaPorosity term only influences the hydraulic behaviour of the flow, blocking it from passing.

The thermal behviour is the same for the whole domain, I am searching for a possibility to interpolate the material, considering the solid areas with the properties of aluminium, while the flow areas shall have water properties.

Is there any possibility to do such things in DASimpleTFoam? Or is the new solver "DAHeatTransferFoam" intended to do such things? If i get it right the "alphaPorosity" term does not exist for this solver, correct?

Thanks for your support!

**Comment by friedenhe** [comment]
> The thermal behviour is the same for the whole domain, I am searching for a possibility to interpolate the material, considering the solid areas with the properties of aluminium, while the flow areas shall have water properties.

Yes, it is possible to compute the thermal properties based on the fluid and solid areas. DAFoam does not have such a feature now, so you need to modify the DASimpleTFoam source code and add a function to interpolate. "DAHeatTransferFoam" is very similar to "DALaplacianFoam", so it is not directly usable for your case.
